src/Story.ts

```diff
@@ -171,17 +172,42 @@ export class Story extends InkObject{
 		// ------
 	}
 
-	public ToJsonString(){
-		let rootContainerJsonList = JsonSerialisation.RuntimeObjectToJToken(this._mainContentContainer);
+	public ToJson(writer?: SimpleJson.Writer) {
+		if (!writer) {
+			writer = new SimpleJson.Writer();
+		}
+
+		writer.WriteObjectStart();
+
+		writer.WriteIntProperty('inkVersion', this.inkVersionCurrent);
+
+		writer.WriteProperty('root', (w) => JsonSerialisation.WriteRuntimeContainer(w, this._mainContentContainer));
+
+		if (this._listDefinitions != null) {
+			writer.WritePropertyStart('listDefs');
+			writer.WriteObjectStart();
+
+			for (let def of this._listDefinitions.lists) {
+				writer.WritePropertyStart(def.name);
+				writer.WriteObjectStart();
 
-		let  rootObject: JObject = {};
-		rootObject['inkVersion'] = this.inkVersionCurrent;
-		rootObject['root'] = rootContainerJsonList;
+				for (let [key, value] of def.items) {
+					let item = InkListItem.fromSerializedKey(key);
+					let val = value;
+					writer.WriteIntProperty(item.itemName, val);
+				}
+
+				writer.WriteObjectEnd();
+				writer.WritePropertyEnd();
+			}
+
+			writer.WriteObjectEnd();
+			writer.WritePropertyEnd();
+		}
 
-		if (this._listDefinitions != null)
-			rootObject['listDefs'] = JsonSerialisation.ListDefinitionsToJToken(this._listDefinitions);
+		writer.WriteObjectEnd();
 
-		return JSON.stringify(rootObject);
+		return writer.ToString();
```

Comment:
Since we don't quite mirror the C# version here (we always return a string, the C# version has a definition that has a void return), maybe we could add a comment to make it explicit?